addons/mail/static/src/models/message/message.js

```diff
@@ -340,8 +340,8 @@ function factory(dependencies) {
             if (this.tracking_value_ids.length > 0) {
                 return false;
             }
-            if (this.originThread.model === 'mail.channel') {
-                return this.message_type === 'comment';
+            if (this.message_type !== 'comment') {
+                return false;
             }
             return this.is_note;
```

Comment:
I think the fix is safe and correct with how `message_type === 'comment'` is being used.
Is the condition on `this.is_note` still relevant? I don't think so.

Actually isn't buggy to keep it based on these changes?
Because at a glance it wouldn't be possible to edit most messages in channels from this change.

Reply:
message type is: email, comment, notification, user_notification

It's not related to note, note is subtype, which should have main type as comment too.

Reply:
But I see what you mean, the ending condition should be "true" for channel, not "is note".

Reply:
I have updated the condition :)